### DISCUSSION TITLE
endpoint to get a .png badge/shield with the generation/co2 of a zone

### DISCUSSION THREAD

**Discussion by Kezii** [electricitymaps/electricitymaps-contrib#4935] [Ideas]
feature request: an endpoint in the app that returns a .png with the details of a zone, like this

![immagine](https://user-images.githubusercontent.com/3357750/211121748-80c50787-04d7-44e8-9a76-ac5134ff28ce.png)

it should return a .png in the same style of the app, with the same data, etc
to integrate it in another website / bot etc

**Comment by VIKTORVAV99** [comment]
I don't think we will ever create a api endpoint that returns a static image, it would be lacking too much information like data sources, carbon intensities, etc.

On the other hand creating a interactive widget could be cool but I'll be honest and say it's low in the list of priorities at the moment.

Since this is not really an issue I will be converting this to a discussion, I hope you understand.

**Comment by madsnedergaard** [comment]
I agree with @VIKTORVAV99 here, but if you want you could actually build something like this yourself using https://co2signal.com - it's only giving CO2 intensity for now, but maybe that's good enough for a badge/shield?

**Comment by Kezii** [comment]
I wanted some way to have the "official graphics" of electricitymap but in an easy to integrate format, otherwise I would use my country's TSO api to get the data

an alternative would be fixing the web app to show the zone details from the url, but with webgl disabled, so I can screenshot the electricitymaps app with an headless browser (try that with webgl disabled, it doesn't work)

if I have to make my own chart then it's not electricitymaps

I get that this is a niche usecase

**Reply by VIKTORVAV99** [reply]
If you need more detailed data than co2signal provides like @madsnedergaard suggested we do have the [commercial API](https://api-portal.electricitymaps.com/).

The data would still come from Electricity Maps (just like co2signla) and you would have more flexibility in how it's presented.

I don't think we consider the app broken with WebGL disabled as it's an integral part of the app.
